src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs

```diff
@@ -23,13 +23,10 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.LanguageService;
 
-internal class CSharpSyntaxFacts : ISyntaxFacts
+internal class CSharpSyntaxFacts : AbstractSyntaxFacts, ISyntaxFacts
 {
     internal static readonly CSharpSyntaxFacts Instance = new();
 
-    // Specifies false for trimOnFree as these objects commonly exceed the default ObjectPool threshold
```

Comment:
>// Specifies false for trimOnFree as these objects commonly exceed the default ObjectPool threshold [](http://example.com/codeflow?start=4&length=99)

Feels odd to remove this pool when it's intentionally avoiding trimming for larger lists.

Reply:
yeah.  i made a call that i don't believe this.  the size of these pools is 128 elements.  So we'd need to commonly see types with more than that many members.  For example, checking with roslyn, i find only 488 types (out of 47057) that have more than that.  So it's truly 1%.  I don't know if we need to not be trimming because of that 1% case.

Reply:
It showed up in the speedometer tests a while back. Might be worth doing a test insertion on this to see if it shows back up again with the pool removed.

PR where the pool was added: [https://github.com/dotnet/roslyn/pull/74790](https://github.com/dotnet/roslyn/pull/74790)